webapp/components/integrations/components/installed_oauth_app.jsx

```diff
@@ -4,70 +4,75 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-import FormError from 'components/form_error.jsx';
-
 import * as Utils from 'utils/utils.jsx';
 
 import {FormattedMessage, FormattedHTMLMessage} from 'react-intl';
-import {regenerateOAuthAppSecret} from 'actions/admin_actions.jsx';
-
+import FormError from 'components/form_error.jsx';
 import DeleteIntegration from './delete_integration.jsx';
 
 const FAKE_SECRET = '***************';
 
-export default class InstalledOAuthApp extends React.Component {
-    static get propTypes() {
-        return {
-            oauthApp: PropTypes.object.isRequired,
-            onDelete: PropTypes.func.isRequired,
-            filter: PropTypes.string
-        };
+export default class InstalledOAuthApp extends React.PureComponent {
+    static propTypes = {
+
+        /**
+        * The oauthApp data
+        */
+        oauthApp: PropTypes.object.isRequired,
+
+        /**
+        * The function to call when Regenerate Secret link is clicked
+        */
+        onRegenerateSecret: PropTypes.func.isRequired,
+
+        /**
+        * The function to call when Delete link is clicked
+        */
+        onDelete: PropTypes.func.isRequired,
+
+        /**
+        * Set to filter OAuthApp
+        */
+        filter: PropTypes.string
     }
 
     constructor(props) {
         super(props);
 
-        this.handleShowClientSecret = this.handleShowClientSecret.bind(this);
-        this.handleHideClientScret = this.handleHideClientScret.bind(this);
-        this.handleRegenerate = this.handleRegenerate.bind(this);
-        this.handleDelete = this.handleDelete.bind(this);
-
-        this.matchesFilter = this.matchesFilter.bind(this);
-
         this.state = {
             clientSecret: FAKE_SECRET
         };
     }
 
-    handleShowClientSecret(e) {
-        e.preventDefault();
+    handleShowClientSecret = (e) => {
+        if (e && e.preventDefault) {
+            e.preventDefault();
+        }
         this.setState({clientSecret: this.props.oauthApp.client_secret});
     }
 
-    handleHideClientScret(e) {
+    handleHideClientSecret = (e) => {
         e.preventDefault();
         this.setState({clientSecret: FAKE_SECRET});
     }
 
-    handleRegenerate(e) {
+    handleRegenerate = (e) => {
         e.preventDefault();
-
-        regenerateOAuthAppSecret(
-            this.props.oauthApp.id,
+        this.props.onRegenerateSecret(this.props.oauthApp.id).then(
             () => {
-                this.handleShowClientSecret(e);
+                this.handleShowClientSecret();
             },
             (err) => {
                 this.setState({error: err.message});
```

Comment:
The promise will never actually get rejected, so I don't think the error function will ever get called. You could check `state.requests.integrations.updateOAuthApp.error` from redux in the success function to check for an error. We're starting to move the redux actions to always return the error but `regenOAuthAppSecret` doesn't do that yet